Source/WebCore/PAL/pal/system/glib/SleepDisablerGLib.cpp

```diff
@@ -91,8 +91,10 @@ void SleepDisablerGLib::acquireInhibitor()
         g_variant_builder_init(&builder, G_VARIANT_TYPE_VARDICT);
         g_variant_builder_add(&builder, "{sv}", "reason", g_variant_new_string(m_reason.utf8().data()));
         parameters = g_variant_new("(su@a{sv})", "" /* no window */, 8 /* idle */, g_variant_builder_end(&builder));
-    } else
-        parameters = g_variant_new("(ss)", g_get_prgname(), m_reason.utf8().data());
+    } else {
+        const gchar* prgname = g_get_prgname();
+        parameters = g_variant_new("(ss)", prgname ? prgname : "", m_reason.utf8().data());
+    }
```

Comment:
I think we could use `FileSystem::currentExecutablePath()` and extract the basename as a fallback, it tries check where the `/proc/self/exe` symlink points to (on Linux, at least):


```suggestion
    } else if (const gchar* prgname = g_get_prgname()) {
        parameters = g_variant_new("(ss)", prgname, m_reason.utf8().data());
    } else if (const auto executablePath = FileSystem::currentExecutablePath(); !executablePath.isNull()) {
        GUniquePtr<char> executableName(g_path_get_basename(executablePath.data()));
        parameters = g_variant_new("(ss)", executableName.get(), m_reason.utf8().data());
    } else {
        parameters = g_variant_new("(ss)", "", m_reason.utf8().data());
    } 
```